src/Forms/bilinearform.jl

```diff
@@ -0,0 +1,42 @@
+default_index_map = prime
+default_bra_vertex_map(v) = (v, "bra")
+default_ket_vertex_map(v) = (v, "ket")
+default_operator_vertex_map(v) = (v, "operator")
+default_tno_constructor(s::IndsNetwork) = delta_network(s)
+
+struct BilinearForm{V,KetMap,BraMap,OperatorMap,SpaceMap} <:
+       AbstractForm{V,KetMap,BraMap,OperatorMap,SpaceMap}
+  tn::ITensorNetwork{V}
+  bra_vs_map::BraMap
+  ket_vs_map::KetMap
+  operator_vs_map::OperatorMap
+  dual_space_map::SpaceMap
+end
+
+#Needed for implementation
+bra_map(blf::BilinearForm) = blf.bra_vs_map
+ket_map(blf::BilinearForm) = blf.ket_vs_map
+operator_map(blf::BilinearForm) = blf.operator_vs_map
+space_map(blf::BilinearForm) = blf.dual_space_map
+tensornetwork(blf::BilinearForm) = blf.tn
+data_graph_type(::Type{<:BilinearForm}) = data_graph_type(tensornetwork(blf))
+data_graph(blf::BilinearForm) = data_graph(tensornetwork(blf))
+
+function BilinearForm(
+  bra::ITensorNetwork, operator::ITensorNetwork, ket::ITensorNetwork; kwargs...
+)
+  return Form(bra, operator, ket; form_type="Bilinear", kwargs...)
+end
+
+function BilinearForm(
+  bra::ITensorNetwork,
+  ket::ITensorNetwork;
```

Comment:
```suggestion
  bra::AbstractITensorNetwork,
  ket::AbstractITensorNetwork;
```
For example to allow `TTN` inputs.

Reply:
Sounds sensible!